samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,749 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This application is specific to the Nordic nRF52840-PDK board.
+ *
+ * It supports the 4 buttons and 4 LEDs as mesh clients and servers.
+ *
+ * Prior to provisioning, a button inverts the state of the
+ * corresponding LED.
+ *
+ * The unprovisioned beacon uses the device address set by Nordic
+ * in the FICR as its UUID and is presumed unique.
+ *
+ * Button and LED 1 are in the root node.
+ * The 3 remaining button/LED pairs are in element 1 through 3.
+ * Assuming the provisioner assigns 0x100 to the root node,
+ * the secondary elements will appear at 0x101, 0x102 and 0x103.
+ *
+ * It's anticipated that after provisioning, the button clients would
+ * be configured to publish and the LED servers to subscribe.
+ *
+ * If a LED server is provided with a publish address, it will
+ * also publish its status on a state change.
+ *
+ * Messages from a button to its corresponding LED are ignored as
+ * the LED's state has already been changed locally by the button client.
+ *
+ * The buttons are debounced at a nominal 250ms. That value can be
+ * changed as needed.
+ *
+ */
+
+/*
+ * Open Issues
+ *
+ * 1. Should the level skeleton remain or just be noted as still just
+ *    a skeleton?
+ * 
+ */
+
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <nrf.h>
+#include <device.h>
+#include <gpio.h>
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/l2cap.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/mesh.h>
+#include <../../../../../subsys/bluetooth/host/hci_core.h>
+#include <../../../../../subsys/bluetooth/common/log.h>
```

Comment:
This looks worrying. A sample application that intends to be a reference for application writers should not encourage the usage of private header files. One reason I could find for this is the usage of the Bluetooth stack-internal logging macros like BT_DBG(). I think using printk() would be the simplest choice here (note that you'll then need \n at the end of all strings). In fact, you already use printk in a few places so this would make the console output consistent as well.

Another reason for the private includes I could find is the usage of bt_dev and that way getting access to the identity address. I think we need to find some better way to do this. If you can live without the persistency you could just generate a random UUID using e.g. bt_rand().

Reply:
The DBG stuff is gone.

Would you consider a patch for a function to return the identity address?
I can imagine a setup where there are a half dozen or more nodes and I'd like to not have them change their ID each time they get flashed.

Reply:
Actually, we already have an API which will give you the identity address: `bt_le_oob_get_local()`

The primary purpose of this API (as the name suggests) is not to get the identity address, rather the address which remote devices will need to connect to, however with your configuration this is in practice also the identity address.

Reply:
I missed it, Problem solved